src/thunderbots/software/ai/navigator/obstacle/velocity_obstacle.cpp

```diff
@@ -0,0 +1,43 @@
+#include "ai/navigator/obstacle/velocity_obstacle.h"
+Polygon VelocityObstacle::getBoundaryPolygon(const Robot& robot,
+                                             double robot_radius_scaling,
+                                             double velocity_projection_scaling)
+{
+    double radius             = ROBOT_MAX_RADIUS_METERS * robot_radius_scaling;
+    Point velocity_projection = robot.velocity() * velocity_projection_scaling;
+    Point velocity_direction  = robot.velocity().norm();
+
+    if (velocity_projection.len() > radius)
+    {
+        return Polygon(
+            {// sides of robot
+             robot.position() + radius * velocity_direction.rotate(Angle::quarter()),
```

Comment:
I would recommend replacing all the multiplier operations to resize the vectors with the `Vector::norm(double length)` function (for all instances in this file). The multiplication operation doesn't quite do what you expect.

Reply:
ok